src/sidebar/components/share-annotations-panel.js

```diff
@@ -59,7 +62,7 @@ function ShareAnnotationsPanel({ analytics, toastMessenger }) {
       title={panelTitle}
       panelName={uiConstants.PANEL_SHARE_ANNOTATIONS}
     >
-      {focusedGroup && mainFrame && (
+      {shareURI && uriShareable && focusedGroup && (
```

Comment:
It seems inconsistent that we could have a `shareURI` defined and yet at the same time a `uriShareable` flag being false. One way to avoid this seeming inconsistency might be to move the logic associated with `uriShareable` into the `pageSharingLink` function, so that `pageSharingLink` returns either a usable URL or `null` if the link is not shareable. Tests for `shareURI && uriShareable` could then become just `shareURI`.

However, you'd need to change the logic for the `These annotations cannot be shared` text so that this content didn't appear if `shareURI` isn't set because the document URI simply isn't known yet (ie. `mainFrame.uri` is not set because eg. the PDF is still loading or document hasn't communicated its URI).

Reply:
Thanks for calling this out. I think some of this tangle comes organically out of trying to appease type-checking, indirectly. I'll attempt to clarify.